### ISSUE TITLE
[a11y]: keyboard tabbing does not work properly w/ SidePanel component

### ISSUE THREAD

**Issue by modean** [carbon-design-system/ibm-products#5400]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

2.40.0

### React version

18.3.1

### Automated testing tool and ruleset

manual testing

### Assistive technology

n/a

### Description

If tabbing into the `SidePanel` using the keyboard the close icon of the `SidePanel` gets the keyboard focus. The icon thus has the blue focus rectangle. Unfortunately once the close icon has the keyboard focus it is no longer possible to move either backwards or forward in the tab order.

Instead the mouse must be used to place the focus explicitly elsewhere.

### WCAG 2.1 Violation

- [Success Criterion 2.1.2 No Keyboard Trap](https://www.w3.org/WAI/WCAG21/Understanding/no-keyboard-trap.html)
- [Failure of Success Criterion 2.1.2 and Conformance Requirement 5 due to combining multiple content formats in a way that traps users inside one format type](https://www.w3.org/WAI/WCAG21/Techniques/failures/F10)

### Reproduction/example

https://codesandbox.io/p/devbox/tabbingsample-wt3yk3?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clx09ppep00073b6i90e5mb0f%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clx09ppep00023b6ipm7mt6wr%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clx09ppep00043b6ijl1nv34j%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clx09ppep00063b6i2no0ryqe%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clx09ppep00023b6ipm7mt6wr%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx09ppep00013b6iybbos7pg%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%257D%255D%252C%2522id%2522%253A%2522clx09ppep00023b6ipm7mt6wr%2522%252C%2522activeTabId%2522%253A%2522clx09ppep00013b6iybbos7pg%2522%257D%252C%2522clx09ppep00063b6i2no0ryqe%2522%253A%257B%2522id%2522%253A%2522clx09ppep00063b6i2no0ryqe%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx09ppep00053b6idzgn56ip%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2523%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clx09ppep00053b6idzgn56ip%2522%257D%252C%2522clx09ppep00043b6ijl1nv34j%2522%253A%257B%2522id%2522%253A%2522clx09ppep00043b6ijl1nv34j%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx09ppep00033b6i6skcnhf1%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clx09ppep00033b6i6skcnhf1%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D

### Steps to reproduce

This description assumes that the linked CodeSandbox is being used.

1. Open the SidePanel using the button on the Carbon header component
2. Click on `Carbon UI shell w/ Carbon for IBM Products`
3. Tab until the tab focus rectangle is on the close icon of the `SidePanel`
4. Try to tab again
5. The tab focus rectangle should remain on the close icon
6. Try pressing shift+tab
7. The tab focus rectangle should remain on the close icon

Note: shift+tab would go one step back in the tab order where tab would go one step ahead in the tab order.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by makafsal** [comment]
@modean - It is the expected behaviour, there are no other focusable elements inside the SidePanel, and that's why it is behaving like that. The focus is trapped in SidePanel as long as it is open, hence to access other components we've to close the SidePanel and proceed.

**Comment by modean** [comment]
@makafsal I'd like to challange your answer in two ways.

Can your provide clear evidence that this is in line w/ the various a11y guidelines and policies. e.g. is there an accessibility champion available that would agree w/ your verdict?
One the other hand our use case is, that the help panel must stay open cause it contains help text that explains how the user would be able to complete a task. If the user would permanently close the help panel it may be harder to complete a task. Also if the user would have to open and close the help panel subsequently, that would be a bad user experience cause the interaction would be reading the help text, then close the panel, complete the task and finally open the help panel again.

Thus I'd like to challange your verdict, cause to me it looks like as the keyboard trap criterion is violated. e.g. you have to do something to be able continuing using the keyboard.

I understand your verdict in the context where the `SidePanel` opens with a cover atop of the main content. There having to click the close button perfectly makes sense. But since it is possible to configure the `SidePanel` wo/ the cover the main content becomes accessible and thus it breaks the keyboard interaction pattern.

What do you think?

**Comment by makafsal** [comment]
@modean - I totally agree with you, I know that there are no accessibility guidelines to support my answer and also, to some extent, this behaviour contradicts the [**WCAG 2.1 Success Criterion 2.1.2: No Keyboard Trap**](https://www.w3.org/WAI/WCAG21/Understanding/no-keyboard-trap). Also in our documentation, it says that a focus trap is not needed in SlideIn SidePanel.



![Image](https://github.com/carbon-design-system/ibm-products/assets/9197089/314d5b6b-ea78-4616-a14b-82db00472cae)



Considering this as a bug and I'll resolve it ASAP.

Thank you for enlightening me 😄 my mistake, apologise for the inconvenience caused due to this issue.

One thing I've noticed from the CodeSandBox is that you're using SidePanel with Header, it is not recommended, instead, you can use `<HeaderPanel />` if possible. Although still, the above-mentioned issue is valid.

**Comment by modean** [comment]
@makafsal thanks for looking into this and thanks for the pointer wrt/ using `<HeaderPanel />`. I will give it a try.
